src/main/java/uk/gov/hmcts/reform/professionalapi/util/JsonFeignResponseUtil.java

```diff
@@ -34,7 +37,13 @@ private JsonFeignResponseUtil() {
     }
 
     public static ResponseEntity<Object> toResponseEntity(Response response, Object clazz) {
-        Optional<Object>  payload = decode(response, clazz);
+        Optional<Object> payload;
+
+        if (response.status() == 401) {
+            payload = Optional.of(new ErrorResponse(UNAUTHORISED, UNSUCCESSFUL_AUTHENTICATION, now().toString()));
+        } else {
+            payload = decode(response, clazz);
```

Comment:
why is it required to handle status code 401 specifically?..

Reply:
when UP send back a 401 response code to PRD (due to S2S issue) and we try to extract the body of the response we get a NPE as the response does not have a body, only a status code. Therefore, we need to handle the 401 and provide a suitable response body (this is the related ticket: https://tools.hmcts.net/jira/browse/RDCC-1728)